lib/ecto/repo/model.ex

```diff
@@ -181,7 +201,8 @@ defmodule Ecto.Repo.Model do
   end
 
   defp with_transactions_if_callbacks(repo, adapter, model, opts, callbacks, fun) do
-    if Enum.any?(callbacks, &function_exported?(model, &1, 1)) do
+    if Enum.any?(callbacks, &function_exported?(model, &1, 1)) and
```

Comment:
@michalmuskala Due to this change, you no longer need to declare `def transaction` in your adapter. I mean, you probably should, but you can simply raise now saying transactions are not supported.